src/pickledb.rs

```diff
@@ -1044,8 +1044,10 @@ impl PickleDb {
 
 impl Drop for PickleDb {
     fn drop(&mut self) {
-        if let PickleDbDumpPolicy::NeverDump = self.dump_policy {
-        } else {
+        if !matches!(
+            self.dump_policy,
+            PickleDbDumpPolicy::NeverDump | PickleDbDumpPolicy::DumpUponRequest
+        ) {
```

Comment:
I'm surprised no tests were failing due to this change. Can you add a test for it?

Reply:
Whoa, right.  I modified `dump_upon_request_policy_test` accordingly, it was explicitly testing for a dump on `Drop`.

However looking at the past CI runs it appears that none of the tests in this repo are currently getting run against PRs 😱

Reply:
I think tests are running against PRs, but not on the right branch. I think it runs against the `master` branch in the PR fork 🤦‍♂️ 
This part: https://github.com/seladb/pickledb-rs/blob/master/.github/workflows/rust.yml#L3-L7

Should be probably changed to: `on: [push, pull_request]`, which will run against every push and PR.

I wrote this a long time ago, I was new to GH Actions at the time... maybe you can change it?

Reply:
On second thought, never mind. I'll merge this PR and fix this separately